mcs/class/corlib/ReferenceSources/Buffer.cs

```diff
@@ -102,129 +110,20 @@ public static unsafe void MemoryCopy (void* source, void* destination, ulong des
 			var src = (byte*)source;
 			var dst = (byte*)destination;
 			while (sourceBytesToCopy > int.MaxValue) {
```

Comment:
If you are making changes here, remove this loop. Pass the entire intptr/int64 size down to the icall.